### PR TITLE
Add the ability to change color programmatically

### PR DESCRIPTION
**Pull request by thadcodes**
This should resolve issue #18, at least for the HarmonyColorPicker.  It looks like the Classic one should already support programmatically updating itself.

### CODE REVIEW

**Comment by ffgiraldez** [comment]
Hi @thadcodes, sorry for the delay. I will look at those changes, but they look outstanding. Thanks.

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt
Comment (riggaroo) [inline color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt:72]: This shouldn't be changed to a Color at this point, the API should return a HsvColor otherwise color information is lost, as multiple HsvColors map to the same Color value. This will break for users who need individual color parts of information.

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt
Comment (riggaroo) [inline color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt:57]: As mentioned in the other comment, this should be reverted as it'll lose specific color detail.

**Review by riggaroo — CHANGES_REQUESTED** [review]
Thanks so much for the contribution towards fixing the programmatic color change!

Two requests:
1. Can you do the brightness value change in a separate PR as its conflating the PR a bit.
2. The API needs to return a Hsv Color so as not to lose those individual properties, for instance the color black can be represented in many different HsvColors: Right now on launch, the color black is broken with this change as it doesn't allow the picker to move around on the palette. Also its defaulting on launch to have the whole color wheel as black, where as before it was brightness = 1
Thanks a lot!

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt
Comment (thadcodes) [reply color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt:72]: Makes sense.  I did that because I wanted to make the input and value changed output match.  So I'll create one where `color: Color` becomes `hsvColor: HsvColor` instead.

**Comment by thadcodes** [comment]
@riggaroo Updated based on your feedback.  Will open the brightness bar PR as soon as this is merged

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt
Comment (ffranco-godaddy) [inline color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt:69]: We should keep the same naming in both `HarmonyColorPicker` and `ClassicColorPicker`, I don't have a strong opinion about which one to choose, but they must be the same.

**Review by ffranco-godaddy — CHANGES_REQUESTED** [review]
Minimum change for LGTM, great work thanks.

File: README.md
Comment (ffgiraldez) [inline README.md:48]: ﻿﻿We must keep both `HarmonyColorPicker` and `HarmonyColorPicker` the same naming, `onValueChange` or `onColorChange`. I don't have a strong opinion about which one to choose.

**Review by ffgiraldez — CHANGES_REQUESTED** [review]

File: README.md
Comment (thadcodes) [reply README.md:48]: I'm a little unclear, are you talking about the two HarmonyColorPicker functions or did you mean ClassicColorPickerScreen and HarmonyColorPicker needed to use the same naming?

If it's the two versions of HarmonyColorPicker I'll note the reason I changed from `onColorChanged` here to `onValueChanged` is because the passed in field name changed from `color` to `value` to keep the signatures for the `Color` old version and the new version that takes in `HsvColor` instead.   I will note that the version with `onColorChanged` is deprecated with a replace with to make it easy to migrate to the new version for upgraders.

In case the issue is the inconsistency between classic and harmony I'm making a push that changes them to be consistent in the non-deprecated version

Comment (thadcodes) [reply README.md:48]: On second thought I just updated it to use the old naming for both.  I think the better distinction between deprecated and non-deprecated is useful because it can help prevent someone from using the old one by mistake.  But I don't feel anywhere near strongly enough to leave it that way

Comment (ffgiraldez) [reply README.md:48]: in this PR the non deprecated Composables are defined in this way
```kotlin
@Composable
fun ClassicColorPicker(
    modifier: Modifier = Modifier,
    hsvColor: HsvColor = HsvColor.from(Color.Red),
    showAlphaBar: Boolean = true,
    onColorChanged: (HsvColor) -> Unit
) {

@Composable
fun HarmonyColorPicker(
    modifier: Modifier = Modifier,
    harmonyMode: ColorHarmonyMode,
    value: HsvColor = HsvColor.from(Color.Red),
    onValueChanged: (HsvColor) -> Unit
) {
   
```
We should keep the parameter names in sync. 

> On second thought I just updated it to use the old naming for both. I think the better distinction between deprecated and non-deprecated is useful because it can help prevent someone from using the old one by mistake. But I don't feel anywhere near strongly enough to leave it that way

Agree with that please update ClassicColorPicker to match HarmonyColorPicker with value/onValueChange.

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt
Comment (ffgiraldez) [inline color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt:62]: [nit] this comment does not add too much value from a user perspective, the opposite of the comments in ClassicColorPicker.

I suggest removing it(as it was previously) or updating it like the ClassicColorPicker

**Comment by thadcodes** [comment]
Sorry @ffgiraldez I pushed my changes to the wrong branch that's why you are not seeing them, I'll have them up on the right branch.   

I can see the argument on the Color Picker that programmatically updates.  I actually wrote that when I was going to have two versions before just deprecating the old one.  I am fine doing that as the comment should be in the deprecation warning now that they are not both correct versions

**Comment by thadcodes** [comment]
ok @ffgiraldez updated, sorry about the wasted review cycle.  🤦‍♂️

**Comment by ffgiraldez** [comment]
You probably will punch me, but please, could you update the example in the README to match real param names?

**Comment by thadcodes** [comment]
> You probably will punch me, but please, could you update the example in the README to match real param names?

@ffgiraldez LOL no not at all, that's what I get for assuming IntelliJ would rename it here too and not checking.  Fixed and pushed.  Lucky you found that because I also had an error in the example that wasn't just renaming

**Comment by ffgiraldez** [comment]
@thadcodes LGTM now, I need to check why the signing is failing on CI😓

**Review by ffgiraldez — APPROVED** [review]

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt
Comment (thadcodes) [reply color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt]: @riggaroo I believe I fixed this. Please let me know if that isn't the case

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt
Comment (riggaroo) [inline color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorPicker.kt:75]: This is shadowing the color parameter. Maybe rename to `updatedColor`?

File: desktop/src/jvmMain/kotlin/HarmonyColorPickerScreen.kt
Comment (riggaroo) [inline desktop/src/jvmMain/kotlin/HarmonyColorPickerScreen.kt:69]: This doesn't seem to be compiling for desktop, is it working for you on desktop?

**Review by riggaroo — COMMENTED** [review]
Thanks @thadcodes this is much better - its just not compiling on desktop, can you double check that it is for you?

File: desktop/src/jvmMain/kotlin/HarmonyColorPickerScreen.kt
Comment (thadcodes) [reply desktop/src/jvmMain/kotlin/HarmonyColorPickerScreen.kt:70]: I see the issue and have pushed it @riggaroo.  Sorry I didn't notice the desktop runner at all 🤦‍♂️

File: desktop/src/jvmMain/kotlin/HarmonyColorPickerScreen.kt
Comment (riggaroo) [inline desktop/src/jvmMain/kotlin/HarmonyColorPickerScreen.kt:34]:
```suggestion
            mutableStateOf(HsvColor.from(Color.Red))
```

File: app/src/main/java/com/godaddy/android/colorpicker/ClassicColorPickerScreen.kt
Comment (riggaroo) [inline app/src/main/java/com/godaddy/android/colorpicker/ClassicColorPickerScreen.kt:33]:
```suggestion
            mutableStateOf(HsvColor.from(Color.Red))
```

Comment (riggaroo) [reply app/src/main/java/com/godaddy/android/colorpicker/ClassicColorPickerScreen.kt:33]: This will just help with the default preview.

File: desktop/src/jvmMain/kotlin/HarmonyColorPickerScreen.kt
Comment (riggaroo) [reply desktop/src/jvmMain/kotlin/HarmonyColorPickerScreen.kt:34]: This will just help with the preview.

**Comment by riggaroo** [comment]
@thadcodes Thanks - that is fixed! 🎉  One last request - can you update `ClassicColorPickerScreen` for the desktop build as that is using the deprecated version?

**Comment by thadcodes** [comment]
> @thadcodes Thanks - that is fixed! 🎉 One last request - can you update `ClassicColorPickerScreen` for the desktop build as that is using the deprecated version?
@riggaroo sorry for the delayed response.  This latest update should address your concerns

**Review by riggaroo — APPROVED** [review]
Thanks @thadcodes! LGTM

**Comment by ffgiraldez** [comment]
Waiting for #49 to merge this. Sorry for all of this @thadcodes

**Pull request merged**
